nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-restapi-processors/src/main/resources/META-INF/docs/org.apache.nifi.processors.elasticsearch.PutElasticsearchRecord/additionalDetails.html

```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html lang="en">
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<head>
+    <meta charset="utf-8" />
+    <title>PutElasticsearchRecord</title>
+    <link rel="stylesheet" href="../../../../../css/component-usage.css" type="text/css" />
+</head>
+<body>
+    <p>This is a record-aware processor built for ElasticSearch 5 and later. The index and type fields are configured
+    with default values that can be overridden using record path operations that find an index or type value in the
+    record set. The ID and operation type (index, update, upsert or delete) can also be extracted in a similar fashion from
+    the record set. The following is an example of a document exercising all of these features:</p>
+    <pre>
+        {
```

Comment:
When doing an `index` or `update`, would the entire document be put in (including the `metadata` object)? If so, is this always a desired behavior? I can see cases for it (vaguely like SCD Type 2 tables in RDBMS) but in other cases, you might want to grab everything BUT the fields that contain information about the operation to be performed, what do you think?

Reply:
Do you know how to remove with JSON Path?

Reply:
For type and index, I added a check for both of them being missing because the way it works is that the attribute version is the default and the jsonpath version overrides that if it is set and evaluates successfully (that's documented in the processor).

Will get to the record version some time tonight hopefully.

Reply:
Appears to already be working. See flow attached at the bottom of this PR.